pkg/quota/v1/generic/evaluator.go

```diff
@@ -33,15 +34,74 @@ import (
 // InformerForResourceFunc knows how to provision an informer
 type InformerForResourceFunc func(schema.GroupVersionResource) (informers.GenericInformer, error)
 
-// ListerFuncForResourceFunc knows how to provision a lister from an informer func
+// ListerFuncForResourceFunc knows how to provision a lister from an informer func.
+// The lister returns errors until the informer has synced.
 func ListerFuncForResourceFunc(f InformerForResourceFunc) quota.ListerForResourceFunc {
 	return func(gvr schema.GroupVersionResource) (cache.GenericLister, error) {
 		informer, err := f(gvr)
 		if err != nil {
 			return nil, err
 		}
-		return informer.Lister(), nil
+		synced := &atomic.Value{}
```

Comment:
why the more complex atomic.Value?  Why not close over a mutex?

Reply:
goal was to avoid permanently contending on informer.HasSynced() once it goes true, so this ratchets a local boolean to true in a threadsafe way